### PR TITLE
FIX: do not include contact url & email in client site settings payload

### PR DESCRIPTION
**Pull request by techAPJ**

### CODE REVIEW

**Review by eviltrout — APPROVED** [review]

**Comment by SamSaffron** [comment]
wait... are we sure about this? 


https://github.com/discourse/discourse/blob/7a0e5c8cb945f2c2aafa314ff71ef4dd2d87899c/lib/guardian.rb#L528-L530

**Comment by techAPJ** [comment]
> wait... are we sure about this?
> 
> https://github.com/discourse/discourse/blob/7a0e5c8cb945f2c2aafa314ff71ef4dd2d87899c/lib/guardian.rb#L528-L530

Note that `contact_url` and `contact_email` are outside `stats` so `can_see_about_stats` will not be applicable on them. I'll add a new guardian check to ensure if contact details can be accessed by client.

File: spec/components/guardian_spec.rb [hunk 1]
```diff
@@ -3830,4 +3830,34 @@
       end
     end
   end
+
+  describe "can_see_site_contact_details" do
```
Comment (tgxworld) [inline spec/components/guardian_spec.rb:3834]:
```suggestion
  describe "#can_see_site_contact_details" do
```

File: app/serializers/about_serializer.rb [hunk 2]
```diff
@@ -35,4 +37,24 @@ def include_stats?
   def stats
     object.class.fetch_cached_stats || Jobs::AboutStats.new.execute({})
   end
+
+  def can_see_site_contact_details
```
Comment (tgxworld) [inline app/serializers/about_serializer.rb:41]: minor but let us make this method private

File: app/serializers/about_serializer.rb [hunk 3]
```diff
@@ -35,4 +37,24 @@ def include_stats?
...
+    scope.can_see_site_contact_details?
```
Comment (tgxworld) [inline app/serializers/about_serializer.rb:42]: I don't think the logic should be pushed into the guardian.. yet. Contact details is very specific to this serializer so moving the logic into guardian makes it more confusing IMO.

**Comment by techAPJ** [comment]
Thanks for the review @tgxworld! PR has been updated per feedback.

**Pull request merged**